packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -57,17 +49,13 @@ export const useGridFilter = (
   const lastFilteringMethodApplied = React.useRef<GridFilteringMethod | null>(null);
 
   useGridStateInit(apiRef, (state) => {
-    if (props.filterModel) {
-      checkFilterModelValidity(props.filterModel);
-    }
+    const filterModel =
+      props.filterModel ?? props.initialState?.filter?.filterModel ?? getDefaultGridFilterModel();
 
     return {
       ...state,
       filter: {
-        filterModel:
-          props.filterModel ??
-          props.initialState?.filter?.filterModel ??
-          getDefaultGridFilterModel(),
+        filterModel: checkFilterModelValidity(filterModel, props.disableMultipleColumnsFiltering),
```

Comment:
It could be renamed `validatedFilterModel` because now this function dispatch the warnings and returned a cleaned model

Reply:
I went for `sanitizeSortModel` and `sanitizeFilterModel`